lib/fhc.js

```diff
@@ -152,7 +152,7 @@ fhc.applyCommandFunction = function (processArgv, cb) {
   // Only new-style commands can be validated
   if (cmdFn.demand) {
     try {
-      argUtils.validate(yargs, cmdFn);
+      argv = argUtils.validate(yargs, cmdFn);
```

Comment:
@camilamacedo86 As I mentioned in PR #307 , this is actually the real problem - we are not passing the new argv to the functions after this point.